api/spec/requests/spree/api/v2/storefront/checkout_spec.rb

```diff
@@ -439,6 +439,26 @@
     end
   end
 
+  describe 'checkout#remove_store_credit' do
+    let(:order_total) { 500.00 }
+    let(:params) { { order_token: order.token, include: 'payments', fields: { payment: 'state' } } }
+    let(:execute) { post '/api/v2/storefront/checkout/remove_store_credit', params: params, headers: headers }
+    let!(:store_credit) { create(:store_credit, amount: order_total) }
+    let!(:order) { create(:order, user: store_credit.user, total: order_total) }
+
+    before do
+      create(:store_credit_payment_method)
+      Spree::Checkout::AddStoreCredit.call(order: order)
+      execute
+    end
+
+    it_behaves_like 'returns 200 HTTP status'
+
+    it 'returns no valid StoreCredit payment' do
+      expect(json_response['included'].all? { |i| i['attributes']['state'] == 'invalid' }).to eq true
```

Comment:
@caudatecoder maybe we could also filter by `payment_method_name` == `StoreCredit` ?

Reply:
@damianlegawiec I think in this example it's safe as we create a fresh order with no payments